orbit/cmd/orbit/orbit.go

```diff
@@ -962,6 +974,19 @@ func main() {
 			g.Add(desktopRunner.actor())
 		}
 
+		// --end-user-email is only supported on Windows (for macOS it gets the
+		// email from the enrollment profile)
+		if runtime.GOOS == "windows" && c.String("end-user-email") != "" {
+			if orbitClient.GetServerCapabilities().Has(fleet.CapabilityEndUserEmail) {
```

Comment:
@lucasmrod Sarah just pointed me to that PR where you remove Capabilities checks: https://github.com/fleetdm/fleet/pull/15771/files#r1433028680

Does that mean we shouldn't do that cap check here? Are capabilities as a whole become unusable because of the orbit config caching, or is it specifically for token rotation because it is so critical to Fleet Desktop?

Reply:
Am all ears.

In the case of Fleet Desktop I considered the feature old enough to not require such a thing (plus we added the `if` to not run such loops+APIs if `--fleet-desktop` was not set/enabled). There would be the edge case of customers running a very old version of Fleet (that doesn't support Fleet Desktop) AND using fleetd with `--fleet-desktop`. Such edge case would see unnecessary API calls on their deployments.

For future capability checks we should make sure we have reached the server. E.g. by introducing a `capabilitiesFetched bool`. (Otherwise the capabilities map could be empty if Fleet is down during startup of fleetd or if the host was deleted from Fleet, which requires a re-enroll and AFAICS capabilities are not updated if there's an auth error?)

Reply:
In our specific case, I think it's not critical if Fleet is down and sending the email cannot succeed, it would eventually work when orbit is started with Fleet available. But I agree with your general idea, for actions that would be done regularly, we should make sure the capabilities have been loaded before considering that the server does not support them.